deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/conf/layers/Layer.java

```diff
@@ -78,12 +79,18 @@
     protected String layerName;
     protected double dropOut;
     protected List<LayerConstraint> constraints;
+    protected Boolean hasBiasConstraints;
+    protected Boolean hasWeightConstraints;
+    protected Set<String> constraintParamNames;
```

Comment:
I don't like this. The parameter names that the constraint applies to should be in the constraints themselves, not on the layer. More generally we should have only *one* field related to constraints, with all relevant information contained within it.